lib/TPP/Dialect/Xsmm/XsmmUtils.cpp

```diff
@@ -564,6 +570,135 @@ FailureOr<SmallVector<Attribute>> getBrgemmFlags(PatternRewriter &rewriter,
   return attributes;
 }
 
+static llvm::SmallDenseSet<int64_t>
+findIndexingOperand(AffineMap indexingMap,
+                    ArrayRef<mlir::utils::IteratorType> iterators,
+                    mlir::utils::IteratorType iter) {
+  assert(iterators.size() == indexingMap.getNumDims());
+  llvm::SmallDenseSet<int64_t> res;
+  for (AffineExpr e : indexingMap.getResults()) {
+    int position = -1;
+    if (isa<AffineDimExpr>(e)) {
+      auto expr = dyn_cast<AffineDimExpr>(e);
+      position = expr.getPosition();
+    } else if (isa<AffineBinaryOpExpr>(e)) {
+      auto lhs = dyn_cast<AffineBinaryOpExpr>(e).getLHS();
+      assert(isa<AffineDimExpr>(lhs));
+      position = (dyn_cast<AffineDimExpr>(lhs)).getPosition();
+    }
+    assert(position >= 0);
+    if (iterators[position] == iter &&
+        llvm::count_if(indexingMap.getResults(), [position](AffineExpr e) {
+          return e.isFunctionOfDim(position);
+        }) == 1)
+      res.insert(position);
+  }
+  return res;
+}
+namespace {
+auto par = mlir::utils::IteratorType::parallel;
+auto red = mlir::utils::IteratorType::reduction;
+} // namespace
+
+FailureOr<linalg::ContractionDimensions>
+inferContractionDims(linalg::GenericOp linalgOp) {
```

Comment:
Just realized, this is a copy of the upstream code. This really should not happen.

This fix need to be upstream. Let's stop work in this PR for now, please.

Reply:
I had to copy the upstream code because the upstream code does not get the floordiv dimension if its an affinebinaryopdimexpr. I forgot to discuss this with you.